packages/patternfly-4/react-core/src/components/Pagination/Navigation.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from '@patternfly/patternfly/components/Pagination/pagination.css';
+import { css } from '@patternfly/react-styles';
+import { AngleLeftIcon, AngleDoubleLeftIcon, AngleRightIcon, AngleDoubleRightIcon } from '@patternfly/react-icons';
+import { Button, ButtonVariant } from '../Button';
+
+const propTypes = {
+  lastPage: PropTypes.number,
+  titles: PropTypes.shape({
+    pages: PropTypes.string
+  }),
+  page: PropTypes.number.isRequired,
+  onSetPage: PropTypes.func.isRequired,
+  onNextclick: PropTypes.func,
+  onPreviousClick: PropTypes.func,
+  onFirstClick: PropTypes.func,
+  onLastClick: PropTypes.func
+};
+const defaultProps = {};
+
+const Navigation = ({
+  page,
+  lastPage,
+  titles,
+  onSetPage,
+  onNextClick,
+  onPreviousClick,
+  onFirstClick,
+  onLastClick,
+  className,
+  ...props
+}) => {
+  return (
+    <nav className={css(styles.paginationNav, className)} aria-label="Pagination" {...props}>
+      <Button
+        variant={ButtonVariant.plain}
+        isDisabled={page === 1}
+        aria-label="Go to first page"
+        data-action="first"
+        onClick={(event) => {
+          onFirstClick(event, 1);
+          onSetPage(event, 1)
+        }}
+      >
+        <AngleDoubleLeftIcon />
+      </Button>
+      <Button
+        variant={ButtonVariant.plain}
+        isDisabled={page === 1}
+        data-action="previous"
+        onClick={(event) => {
+          const newPage = page - 1 >= 1 ? page - 1 : 1
+          onPreviousClick(event, newPage);
+          onSetPage(event, newPage);
+        }}
+        aria-label="Go to previous page"
+      >
+        <AngleLeftIcon />
+      </Button>
+      <div className={css(styles.paginationNavPageSelect)}>
+        <input
+          className={css(styles.formControl)}
+          aria-label="Current page"
+          type="number"
+          min="1"
+          max={lastPage}
+          size="2"
+          value={page}
+          onChange={(event) => {
+            let inputPage = Number.parseInt(event.target.value);
+            inputPage = isNaN(inputPage) ? page : inputPage;
+            inputPage = inputPage > lastPage ? lastPage : inputPage;
+            inputPage = inputPage < 1 ? 1 : inputPage;
+            onSetPage(event, isNaN(inputPage) ? page : inputPage)
+          }}
+        />
+        <span aria-hidden="true">of {lastPage} {titles.pages}</span>
```

Comment:
This may be a good use case for the template string method added to helpers/utils.ts. We could pass in the entire string as a prop and use the last page & titles.pages values as embedded expressions.